services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py

```diff
@@ -107,14 +107,21 @@ def _apply_templating_directives(
 
 
 def _merge_env_vars(
-    compose_spec_env_vars: list[str], settings_env_vars: list[str]
+    compose_spec_env_vars: list[str] | dict[str, str],
+    settings_env_vars: list[str] | dict[str, str],
 ) -> list[str]:
     def _gen_parts_env_vars(
-        env_vars: list[str],
+        env_vars: list[str] | dict[str, str],
     ) -> Generator[tuple[str, str], None, None]:
-        for env_var in env_vars:
-            key, value = env_var.split("=")
-            yield key, value
+        assert isinstance(env_vars, list | dict)  # nosec
+
+        if isinstance(env_vars, list):
```

Comment:
THOUGHT: This is the reason why we impose certain standards. The env section of our docker-compose are required to be key-value. When you establish certain guarantees at the source, there is no need to be doing these checks. It is better for the Umwelt ;-)